proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,238 @@
+        <dependency id="Microsoft.Windows.CsWin32" version="0.2.138-beta" include="All" />
+      </group>
+    </dependencies>
+```
+
+##### Case 2 for PackageReference
+
+Package reference in csproj file.
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <VersionSuffix>beta</VersionSuffix>
+    <PrivateAssetIndependent>True</PrivateAssetIndependent>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta" PrivateAssets="none" IncludeAssets="none" />    
+  </ItemGroup>
+```
+
+Before change nuspec file:
+
+```.net
+    <dependencies>
+      <group targetFramework=".NETStandard2.0" />
+    </dependencies>
+```
+
+After change nuspec file:
+
+```.net
+    <dependencies>
+      <group targetFramework=".NETStandard2.0">
+        <dependency id="Microsoft.Windows.CsWin32" version="0.2.138-beta" include="All" />
+      </group>
+    </dependencies>
+```
+
+##### Case for Project to Project reference
+
+`ProjectReference` in parent `ClassLibrary1` csproj file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\refissue\refissue.csproj" />
+  </ItemGroup>
+```
+
+And `PackageReference` in `refissue.csproj` file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <VersionSuffix>beta</VersionSuffix>
+    <PrivateAssetIndependent>True</PrivateAssetIndependent>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta" PrivateAssets="none" ExcludeAssets="all" />
+  </ItemGroup>
+```
+
+Before changing, `Microsoft.Windows.CsWin32.props` file in `build` asset doesn't flow to ClassLibrary1.csproj.nuget.g.props file in obj folder.
+
+After change, `Microsoft.Windows.CsWin32.props` file in `build` asset flow to ClassLibrary1.csproj.nuget.g.props file in obj folder.
+
+### Technical explanation
+
+We already have a [logic](hhttps://github.com/NuGet/NuGet.Client/blob/380415d812681ebf1c8aa0bc21533d4710514fc3/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L577-L582) that combines `IncludeAssets/ExcludeAssets` options with `PrivateAssets` for pack operation, we just make logic depending on opt-in property.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+- Might break some consuming projects, that is why it's opt-in feature. See table in #functional-explanation above for more details.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+- We could make it opt-in option in `nuget.config` file, but it doesn't give customer to option to opt in/out for per project level.
```

Comment:
I'm not suggesting add a nuget.config switch, but feel like the best way for customers to use this feature is on either *all* or *none* of their projects.

Reply:
It's been a while since I thought about this issue, but I remember my preference is that it should be opt-in per project, maybe even per-package. 😁 

Frankly, I've forgotten why it's important to be able to exclude an asset from the current project, but still have it flow transitively, so I don't feel like it's a very common scenario, so having it "unexpectedly" in all projects in a repo might not be desirable.

Reply:
It's significantly simpler for it to be on a per project level. 

The calculation is not for the PackageReference in question only at pack time, but rather for potential project references as well.

Reply:
Sure, we had a lot of discussion around this. I learned a lot from the discussion, since we already decided on adding `ExcludedAssetsFlow` at per package reference I'm resolving this comment. Still, someone want to onboard the whole project or solution it's doable using msbuild script/syntax.